pandas/core/groupby/generic.py

```diff
@@ -639,7 +640,12 @@ def value_counts(
         rep = partial(np.repeat, repeats=np.add.reduceat(inc, idx))
 
         # multi-index components
-        labels = list(map(rep, self.grouper.recons_labels)) + [llab(lab, inc)]
+        if isinstance(self.grouper, BinGrouper) and (
+            len(self.grouper.binlabels) != len(self.grouper.indices)
+        ):
```

Comment:
so this should be done in recons_label itself

Reply:
@jreback Finally, I did it with overriding recons_labels method from BaseGrouper in BinGrouper.
```diff
diff --git a/pandas/core/groupby/ops.py b/pandas/core/groupby/ops.py
index 40517eefe..e4335d399 100644
--- a/pandas/core/groupby/ops.py
+++ b/pandas/core/groupby/ops.py
@@ -825,6 +833,14 @@ class BinGrouper(BaseGrouper):
     def names(self):
         return [self.binlabels.name]

+    @property
+    def recons_labels(self):
+        comp_ids, obs_ids, _ = self.group_info
+        if len(self.binlabels) != len(self.indices):
+            return [np.unique(comp_ids)]
+        labels = (ping.labels for ping in self.groupings)
+        return decons_obs_group_ids(comp_ids, obs_ids, self.shape, labels, xnull=True)
+
     @property
     def groupings(self):
         from pandas.core.groupby.grouper import Grouping
```

However, after I fetch upstream to this branch, a lot of files are changed. I don't know whether it is okay. I'll really appreciate it if you tell me what should I do for this.